src/ota_image_libs/v1/file_table/db.py

```diff
@@ -197,36 +201,39 @@ class AsyncFileTableResourceORMPool(
 #
 # ------ Helper for operating the file_table DB ------ #
 #
+# NOTE: although we define each helper classes as tables,
+#       we only need the de-serialization feature, not actually
+#       to create database against the table defs.
 
 
-class _FileTableEntryTypedDict(TypedDict):
+class _FileTableEntry(TableSpec):
     """The result of joining ft_inode and ft_* table."""
 
-    path: str
-    uid: int
-    gid: int
-    mode: int
-    links_count: Optional[int]
-    xattrs: Optional[MsgPackedDict]
+    path: Annotated[str, SkipValidation]
+    uid: Annotated[int, SkipValidation]
+    gid: Annotated[int, SkipValidation]
+    mode: Annotated[int, SkipValidation]
+    links_count: Annotated[Optional[int], SkipValidation] = None
+    xattrs: Annotated[Optional[MsgPackedDict], TypeAffinityRepr(bytes)] = None
```

Comment:
Good. Maybe the intention is to skip the validation from database for performance.

Reply:
Yes, as these data types are sqlite3 native data types, sqlite3 can do the de-serialization for us. However, the `xattrs` field is stored as `bytes` in the database, actually it is msgpacked msg, which we need to further de-serialization into `MsgPackedDict` object.